esmvalcore/preprocessor/__init__.py

```diff
@@ -463,10 +465,34 @@ def save(self):
     def close(self):
         """Close the file."""
         if self._cubes is not None:
+            self.update_attributes()
             self.save()
             self._cubes = None
             self.save_provenance()
 
+    def update_attributes(self):
+        """Update product attributes from cube metadata."""
+        if not self._cubes:
+            return
```

Comment:
I would rather leave that here. It's a public function, so in theory it could also be called at other places. Also, the check ensures that the CubeList is not empty.

Reply:
I'm not too happy about PreprocessorFile as part of the public API because I think it isn't well enough designed for that. I never intended it to be public, this is an accident, that's why it's not in `__all__` and not documented. Could you just add the new method as a private method, so at least we're not extending this unwanted public API?

Reply:
I created an issue to discuss this: #1847

Reply:
Manu marked this as resolved, which one is it - all good by you @bouweandela and things will be treated in a separate PR to close that issue?